core/modules/taxonomy/taxonomy.entity.inc

```diff
@@ -104,6 +104,66 @@ class TaxonomyTerm extends Entity {
       'options' => array(),
     );
   }
+
+  /**
+   * Overrides Entity::createAccess().
+   */
+  public static function createAccess($bundle = NULL, $account = NULL) {
+    $rights = &backdrop_static(__METHOD__, array());
+
+    // If no user object is supplied, the access check is for the current user.
+    if (empty($account)) {
+      $account = $GLOBALS['user'];
+    }
+
+    // If we've already checked access for this node, user and op, return from
+    // cache.
+    if (isset($rights[$account->uid][$bundle])) {
+      return $rights[$account->uid][$bundle];
+    }
+
+    $rights[$account->uid][$bundle] = user_access('administer taxonomy', $account);
+    return $rights[$account->uid][$bundle];
```

Comment:
Same as with `File::access()` functions: if our checking is so simple as just calling `user_access()` we can skip the caching mechanism.